core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/toDataFrame.kt

```diff
@@ -28,6 +28,11 @@ public inline fun <reified T> Iterable<T>.toDataFrame(): DataFrame<T> =
         properties()
     }
 
+public fun Iterable<*>.toDataFrame(klass: KClass<*>): DataFrame<*> =
```

Comment:
I'd add some small KDocs for new public functions :) Also, maybe people want to specify roots/maxdepth as well, would that be possible here?

Reply:
Also, since we supply a `KClass<T>` we can get the type information. If we have an `Iterable<Any?>`, we know we'll get a `DataFrame<T>`.

Reply:
Finally, would it make sense to offer overloads with `body: CreateDataFrameDsl<T>.() -> Unit` too?

Reply:
> Also, since we supply a KClass<T> we can get the type information. If we have an Iterable<Any?>, we know we'll get a DataFrame<T>.
If you have KClass with specific T, i'd say you might as well use list.toDataFrame() without KClass?

> Finally, would it make sense to offer overloads with body: CreateDataFrameDsl<T>.() -> Unit too?
Well, this overload is `CreateDataFrameDsl<T>.() -> Unit = { properties() }` so we'd need to combine them. Also it's supposed to be used when only runtime KClass<*> is available. What useful can be done with `CreateDataFrameDsl<Any?>.() -> Unit`?

> Also, maybe people want to specify roots/maxdepth as well, would that be possible here?
roots - no, because Any?. maxDepth - yes

Reply:
> If you have KClass with specific T, i'd say you might as well use list.toDataFrame() without KClass?

Hmm, I guess you're right. I first thought it might be possible to use another class as argument of `toDataFrame()`, then it would make sense to get `T`... However, that results in `java.lang.IllegalArgumentException: object is not an instance of declaring class`. It might be worth mentioning in attached KDocs that all elements in the iterable must be exact instances the KClass given.